### PR TITLE
Add Info to config and use PostBuildSwaggerObjectHandler

### PR DESCRIPTION
**Pull request by wsong**
Also added a GetSwaggerSpec function, which is useful when you just want to
generate an offline JSON blob for the Swagger spec

### CODE REVIEW

**Comment by emicklei** [comment]
I am still not sure about adding the Info object to the config struct (as it was in swagger12). If you look at the example you see that the post build handler is used to fill in the Swagger object. This gives the developer all the freedom it needs.

**Comment by emicklei** [comment]
I propose a rename of the new function.
The "Get" prefix is unusual in Go and it wronly suggests that there is a SwaggerSpec value to get.
Furthermore, it is a Swagger object that is returned.
A better name would be BuildSwagger(config) *Swagger

File: spec_resource.go [hunk 1]
```diff
@@ -33,10 +43,10 @@ func NewOpenAPIService(config Config) *restful.WebService {
 			Definitions: definitions,
 		},
 	}
-	resource := specResource{swagger: swagger}
-	ws.Route(ws.GET("/").To(resource.getSwagger))
-
-	return ws
+	if config.PostBuildSwaggerObjectHandler != nil {
```
Comment (rmohr) [inline spec_resource.go]: Isn't PostBuildSwaggerObjectHandler then called twice?

Comment (emicklei) [inline spec_resource.go]: where do you see the second invocation of the handler with the Swagger object?

Comment (rmohr) [inline spec_resource.go]: I was working under the assumption that it was already somewhere else called before (e.g. in go-restful), but if it was nowhere called until this PR, then it is just called once ...

**Pull request merged**